### PR TITLE
"Edit" button on code examples

### PR DESCRIPTION
**Pull request by PuruVJ**
Based on #35188, this PR adds the "Edit" button on code examples, clicking which will open up stackblitz editor in new tab with the code example in mind.

Here's a video:

https://user-images.githubusercontent.com/47742487/147886526-f8ed5544-d675-49b4-a6d6-20d211a7fc53.mp4

### CODE REVIEW

**Comment by XhmikosR** [comment]
Hey, thanks for the PR, but this has many unrelated changes.

**Comment by PuruVJ** [comment]
Can you specify the unrelated changes? So I can rectify them

**Comment by XhmikosR** [comment]
Sorry, I don't have the time to clarify everything. For starters you are not running the lint scripts locally.

Secondly, you are loading the SDK from unpkg. You should use a real CDN and pin the version. And finally, you have hardcoded stuff.

**Comment by PuruVJ** [comment]
Changed it from unkpg to jsdelivr

I ran the lint scripts and fixed the issue with EOF on _clipboard-js.scss

If you have time, can you specify what you mean by "hard-coded"?

File: site/layouts/partials/scripts.html [hunk 1]
```diff
@@ -18,3 +18,39 @@
 {{- end }}
 
 <script src="{{ $docsJs.Permalink | relURL }}"></script>
+
+
+{{ printf .Site.Params.cdn.css | safeHTMLAttr }}
```
Comment (XhmikosR) [inline site/layouts/partials/scripts.html:23]: This is unrelated.

File: site/layouts/partials/scripts.html [hunk 2]
```diff
@@ -18,3 +18,39 @@
...
+  StackBlitzSDK.openBootstrapSnippet = function(snippet) {
+
+    const project = {
+        files: {
+          "index.html": `<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">
```
Comment (XhmikosR) [inline site/layouts/partials/scripts.html:36]: These are hardcoded

File: site/layouts/shortcodes/example.html [hunk 3]
```diff
@@ -22,6 +22,9 @@
 {{- end -}}
 
 {{- if eq $show_markup true -}}
+  <div class="bd-edit">
+    <button type="button" class="btn-edit" title="Edit on StackBlitz">Edit</button>
```
Comment (XhmikosR) [inline site/layouts/shortcodes/example.html:26]: It seems to me you are reinventing the wheel with this button's styles.

File: site/assets/js/application.js [hunk 4]
```diff
@@ -4,8 +4,8 @@
 
 /*!
  * JavaScript for Bootstrap's docs (https://getbootstrap.com/)
- * Copyright 2011-2021 The Bootstrap Authors
- * Copyright 2011-2021 Twitter, Inc.
+ * Copyright 2011-2022 The Bootstrap Authors
```
Comment (XhmikosR) [inline site/assets/js/application.js:7]: Unrelated changes here

File: site/assets/js/application.js [hunk 5]
```diff
@@ -169,4 +169,43 @@
     icon: '#'
   }
   anchors.add('.bd-content > h2, .bd-content > h3, .bd-content > h4, .bd-content > h5')
+
+  // Open in StackBlitz logic
+  /*
```
Comment (XhmikosR) [inline site/assets/js/application.js:174]: No need to keep commented out code

File: site/layouts/partials/scripts.html [hunk 6]
```diff
@@ -18,3 +18,39 @@
 {{- end }}
 
 <script src="{{ $docsJs.Permalink | relURL }}"></script>
+
+
+{{ printf .Site.Params.cdn.css | safeHTMLAttr }}
+
+<script src="https://cdn.jsdelivr.net/npm/@stackblitz/sdk@1.5.3/bundles/sdk.umd.min.js"></script>
```
Comment (XhmikosR) [inline site/layouts/partials/scripts.html:25]: We don't need to load this in every page. Which probably means, you shouldn't add the logic in application.js.

**Comment by PuruVJ** [comment]
Hi, I will close this PR and make a new one with new changes. I am having some issues in git locally

**Comment by PuruVJ** [comment]
New PR here #35644

File: site/layouts/partials/scripts.html (hunk 1 above)
Comment (PuruVJ) [reply site/layouts/partials/scripts.html:23]: Fixed in the new PR

File: site/layouts/partials/scripts.html (hunk 2 above)
Comment (PuruVJ) [reply site/layouts/partials/scripts.html:36]: Fixed in the new PR

File: site/layouts/shortcodes/example.html (hunk 3 above)
Comment (PuruVJ) [reply site/layouts/shortcodes/example.html:26]: Fixed in the new PR

File: site/assets/js/application.js (hunk 4 above)
Comment (PuruVJ) [reply site/assets/js/application.js:7]: Fixed in the new PR

File: site/layouts/partials/scripts.html (hunk 6 above)
Comment (PuruVJ) [reply site/layouts/partials/scripts.html:25]: Which file do you suggest I add this logic?

File: site/assets/js/application.js (hunk 5 above)
Comment (PuruVJ) [reply site/assets/js/application.js:174]: Fixed in Original PE

File: site/layouts/partials/scripts.html (hunk 6 above)
Comment (PuruVJ) [reply site/layouts/partials/scripts.html:25]: Would adding a script tag to the `example.html` partial and putting "open in stackblitz" logic there be fine?
